src/main/java/net/neoforged/neoforge/common/crafting/SizedIngredient.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.crafting;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.network.codec.ByteBufCodecs;
+import net.minecraft.network.codec.StreamCodec;
+import net.minecraft.util.ExtraCodecs;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.Ingredient;
+import net.neoforged.neoforge.common.util.NeoForgeExtraCodecs;
+
+/**
+ * Standard implementation for an ingredient and a count.
+ * {@link Ingredient} does not perform count checks, so this class is used to wrap an ingredient with a count,
+ * and provide a standard serialization format.
+ */
+public record SizedIngredient(Ingredient ingredient, int count) {
```

Comment:
Some static constructors that mirror ingredient might be nice to have, e.g. `SizedIngredient.of(TagKey<Item> tag, int count)` or `SizedIngredient.of(int count, Item... item)`. Not strictly needed but saves a level of nesting on object creation.

Reply:
added